Src/coffeescript/cql-execution/src/elm/interval.coffee

```diff
@@ -133,4 +133,36 @@ module.exports.Starts = class Starts extends UnimplementedExpression
 
 module.exports.Ends = class Ends extends UnimplementedExpression
 
-module.exports.Collapse = class Collapse extends UnimplementedExpression
+module.exports.Collapse = class Collapse extends Expression
+  constructor: (json) ->
+    super
+
+  exec: (ctx) ->
+    result = @execArgs ctx
+    if result?.length > 1
+      # we don't handle imprecise intervals at this time
+      for a in result
+        if a.low.isImprecise?() || a.high.isImprecise?()
+          throw new Error("Collapse does not support imprecise dates at this time.")
+
+      # sort intervals by start
+      result.sort (a,b)->
+        return -1 if a.low < b.low
+        return 1 if a.low > b.low
+        0
+
+      # collapse intervals as necessary
+      collapsedIntervals = result
+      result = []
+      a = collapsedIntervals.shift()
+      b = collapsedIntervals.shift()
+      while b
+        if b.low <= a.high
```

Comment:
Again, those native `<=` operators shouldn't work if this is a `DateTime`.  For `<=` you'd have to use `sameOrBefore`.  Not sure why your tests are passing -- but it's probably just luck (or you're passing in the intervals in the right order already).